Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -183,23 +185,38 @@ sub handleFeed {
 			playlist => $url,
 		};
 
+		# if pre-cached feed data is missing, initiate retrieval
 		unless ($image && $cache->get('podcast_moreInfo_' . $url)) {
-			Slim::Formats::XML->getFeedAsync(
-				sub {
-					_precacheShowDetails($url, $_[0]);
-				},
-				sub {
-					$log->warn("can't get $url RSS feed information: ", $_[0]);
-				},
-				{
-					parser => 'Slim::Plugin::Podcast::Parser',
-					url => $_->{value},
-					expires => 86400
-				}
-			);
+			# cache a placeholder image & moreInfo to guard against retrieving
+			# the feed multiple times while browsing within the podcast menu
+			# they will be replaced with real data after feed is successfully retrieved
+			$cache->set('podcast-rss-' . $url, __PACKAGE__->_pluginDataFor('icon'), '1days');
+			$cache->set('podcast_moreInfo_' . $url, {}, '1days');
+			push (@need, $url);
 		}
 	}
 
+	# get missing cache images & moreinfo if any
+	# each feed is retrieved and parsed sequentially, to limit loading on modestly powered servers
+	$fetch = sub {
+		my $url = pop @need;
+		Slim::Formats::XML->getFeedAsync(
+			sub {
+				precacheFeedData($url, $_[0]);
+				$fetch->();
+			},
+			sub {
+				$log->warn("can't get $url RSS feed information: ", $_[0]);
+				$fetch->();
+			},
+			{
+				parser  => 'Slim::Plugin::Podcast::Parser',
+				url     => $url,
+			}
+		) if $url;
```

Comment:
Do you even need to call `precacheFeedData()` here? Wouldn't that be called from the parser already? Maybe that was what caused an error twice?

Reply:
You are right, we don't need to call it here because the parser does it. Will amend.

But I know it wasn't the cause of calling the error callback twice. That was triggered by a connection problem.
First callback reported: "Connect timed out: Connection timed out".
Second callback reported: "Connect timed out: Bad file descriptor".
 
Possibly at some level the http connection object was not being properly released.